src/main/java/io/vertx/lang/jruby/JRubyVerticle.java

```diff
@@ -0,0 +1,119 @@
+package io.vertx.lang.jruby;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Verticle;
+import io.vertx.core.Vertx;
+import org.jruby.CompatVersion;
+import org.jruby.RubyClass;
+import org.jruby.RubyInstanceConfig;
+import org.jruby.embed.LocalContextScope;
+import org.jruby.embed.ScriptingContainer;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class JRubyVerticle implements Verticle {
+
+  private final ClassLoader classLoader;
+  private final String verticleName;
+  private Vertx vertx;
+  private Context context;
+  private ScriptingContainer container;
+
+  public JRubyVerticle(ClassLoader classLoader, String verticleName) {
+    this.classLoader = classLoader;
+    this.verticleName = verticleName;
+  }
+
+  @Override
+  public Vertx getVertx() {
+    return vertx;
+  }
+
+  @Override
+  public void init(Vertx vertx, Context context) {
+    this.vertx = vertx;
+    this.context = context;
+  }
+
+  @Override
+  public void start(Future<Void> startFuture) throws Exception {
+    container = new ScriptingContainer(LocalContextScope.SINGLETHREAD);
```

Comment:
A new ScriptingContainer per verticle instance could be very memory hungry. Originally we did it this way in the old Vert.x 2.0 ruby impl, but moved to a ScriptingContainer per Vert.x instance as we could only deploy a couple hundred ruby verticles otherwise.